test/endpoints/googlenow/googlenow_test.go

```diff
@@ -311,8 +316,8 @@ func TestAddCredentialsMissingAuthCode(t *testing.T) {
 		return
 	}
 
-	assert.Equal(t, "400 Bad Request", res.Status)
-	assert.Equal(t, 0, counter)
+	assert.Equal(t, "500 Internal Server Error", res.Status)
+	assert.Equal(t, 1, counter)
```

Comment:
This is the same as before. It's missing a required field which will return 500 instead of 400 in the encoding/json world since there is no validation.

Reply:
I would consider this a major version bump since we are changing the status codes.

Reply:
Sure, I agree this can be a version bump. 

However, to clarify, any clients that use easyjson will continue to see the same behavior as before. Also, similarly, any clients that continue to use encoding/json will continue to see the same behavior. This is because the UnamrshalBody() function takes in an interface to unmarshal: 

`func (req *ServerHTTPRequest) UnmarshalBody(
	body interface{}, rawBody []byte,
)`

The status code is changing here only because we are moving from easyjson to encoding/json for `example-gateway` endpoints only.

Reply:
The cause for concern is the status code change when no interfaces have changed, I won't expect the runtime behavior to change as well.

Reply:
This is a big functionality change by switching away from easyjson. Easyjson recognizes `required` tag in Go struct and generates code accordingly to make sure payload has the required fields, otherwise encoding/decoding fails. It is essentially payload validation, and we will lose that functionality and the `required` modifier in thrift or tag in Go struct ends up being no-op. 

The concern that @rpatali raised is valid, this is not only an API change but also has safety implications, i.e. previously invalid payload, which was correctly recognized as bad request and stopped from further propagation, will now flow thru the stack. The consequence is that NPE could happen anywhere unless deliberate validations are in place.

Reply:
Thanks @ChuntaoLu. We will use validation tags (https://github.com/go-playground/validator) to preserve this from happening in the real world. We don't use the validation library today in Zanzibar, however it can be added to mimic the same behavior.